tests/fixtures/optimizer/expand_laterals.sql

```diff
@@ -0,0 +1,40 @@
+# title: expand alias reference
+SELECT
+  x.a + 1 AS i,
+  i + 1 AS j,
+  j + 1 AS k
+FROM x;
+SELECT
+  x.a + 1 AS i,
+  x.a + 1 + 1 AS j,
+  x.a + 1 + 1 + 1 AS k
+FROM x;
+
+# title: noop - reference comes before alias
+SELECT
+  b + 1 AS j,
+  x.a + 1 AS i
+FROM x;
+SELECT
+  b + 1 AS j,
+  x.a + 1 AS i
+FROM x;
+
+
+# title: subquery
+SELECT
+  *
+FROM (
+  SELECT
+    x.a + 1 AS i,
+    i + 1 AS j
+  FROM x
+);
+SELECT
+  *
+FROM (
+  SELECT
+    x.a + 1 AS i,
+    x.a + 1 + 1 AS j
+  FROM x
+);
```

Comment:
what about the case where it's ambiguous, we need to default to the table and not the alias, is that tested

select x.b AS a,
a
FROM x

a should be the table

Reply:
seems like it's the table column, I also tried that now

Reply:
it's because `expand_laterals` runs after `qualify_columns` and the replacement condition in the former involves checking if there's a table in the column expression.

Reply:
That's right, this implicitly depends on qualify columns.

I could make this independent, but that'll require building scope. I'm not sure what the scope performance overhead is, but in general I try to avoid constructing it unnecessarily.. 

Happy to switch.

Reply:
no i was just asking for a test